src/components/App/appReducer.js

```diff
@@ -1,118 +1,206 @@
-import geoServerXmlReducer from './reducers/geoServerXmlReducer';
-import menuReducer from '../Menu/menuReducer';
+import geoServerXmlReducer from './reducers/geoServerXmlReducer'
+import menuReducer from '../Menu/menuReducer'
 
 const appReducer = (state = [], action) => {
     switch(action.type){
         case 'POPULATE_APP':
             let layers = geoServerXmlReducer(action.xmlData.xmlData);
             layers = layers.map(l => {
-                return {...l,
+                return {
+                    ...l,
                     selected: false,
-                    showDescription: false
+                    match: true,
+                    showDescription: false,
                 }
             });
-            let menuItems = menuReducer(layers);
+            let menuItems = menuReducer(layers)
             menuItems = menuItems.map(m => {
-                return {...m,
-                    selected: false
+                return {
+                    ...m,
+                    selected: false,
+                    match: true,
                 }
             });
             return {
                 currentLevel: 0,
                 layers,
-                menuItems
+                menuItems,
             };
         case 'TOGGLE_LAYER':
-            let newLayers = [];
+            var newLayers = []
```

Comment:
I believe the problem is scope. "let" makes the varible attached to a block. A block, in Javascript, initiates with { and ends with }. As this code is within a switch / case condition, there is only one big block. If I set "let newLayers" in one case, and "let newLayers" on another case, within the same block, it throws an error, because we cannot redeclare block-scoped variable. "var", on the other hand, is function block scoped and can be redeclared within the same function block scope.